File: dotnet/console/Console/Scenarios/CreateDocumentWithEnvelopeScenario.cs
using Lacuna.Signer.Api;
using Lacuna.Signer.Api.Documents;
using Lacuna.Signer.Api.FlowActions;
using Lacuna.Signer.Api.Users;
using Lacuna.Signer.Client;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Console.Scenarios
{
    class CreateDocumentWithEnvelopeScenario : Scenario
    {
        /**
        * This scenario demonstrates the creation of a document with envelope.
        */
        public override async Task RunAsync()
        {
            // 1. The file's bytes must be read by the application and uploaded
            var filePath = "sample.pdf";
            var fileName = Path.GetFileName(filePath);
            var file = File.ReadAllBytes(filePath);
            var uploadModel = await SignerClient.UploadFileAsync(fileName, file, "application/pdf");
            var uploadModelTwo = await SignerClient.UploadFileAsync(fileName, file, "application/pdf");

            // 2. Define the name of the document which will be visible in the application
            var fileUploadModel = new FileUploadModel(uploadModel) { DisplayName = "One Envelope Sample" };
            var fileUploadModelTwo = new FileUploadModel(uploadModelTwo) { DisplayName = "One Envelope Sample" };

            // 3. For each participant on the flow, create one instance of ParticipantUserModel
            var participantUser = new ParticipantUserModel()
            {
                Name = "Jack Bauer",
                Email = "jack.bauer@mailinator.com",
                Identifier = "75502846369"
            };

            // 4. Create a FlowActionCreateModel instance for each action (signature or approval) in the flow.
            //    This object is responsible for defining the personal data of the participant and the type of 
            //    action that he will perform on the flow
            var flowActionCreateModel = new FlowActionCreateModel()
            {
                Type = FlowActionType.Signer,
                User = participantUser
            };

            // 5. Send the document create request setting up the attribute "isEnvelope" as "true" and you MUST give a name to this envelope "EnvelopeName"
            var documentRequest = new CreateDocumentRequest()
            {
                Files = new List<FileUploadModel>() { fileUploadModel, fileUploadModelTwo},
                FlowActions = new List<FlowActionCreateModel>() { flowActionCreateModel },
                IsEnvelope = true,
                EnvelopeName = "Envelope"
            };
            var result = (await SignerClient.CreateDocumentAsync(documentRequest)).First();

            System.Console.WriteLine($"Document {result.DocumentId} created");
        }
    }
}

File: nodejs/scenarios/createDocumentWithEnvelopeScenario.ts
import {
  UploadApi,
  DocumentsApi,
  FileUploadModel,
  UsersParticipantUserModel,
  FlowActionsFlowActionCreateModel,
  FlowActionType,
  getBase64,
} from "signer-node-client";
import { config } from "./scenario";

/**
 * This scenario demonstrates the creation of a document with description.
 */

const filePath = "../../samples/sample.pdf";
const fileName = "sample.pdf";

const uploadApi = new UploadApi(config);
const documentsApi = new DocumentsApi(config);

// 1. The file's bytes must be read by the application and uploaded
uploadApi
  .apiUploadsBytesPost({
    bytes: getBase64(filePath),
  })
  .then((responseOne) => {
    uploadApi
      .apiUploadsBytesPost({
        bytes: getBase64(filePath),
      })
      .then((responseTwo) => {
        // 2. Define the name of the document which will be visible in the application
        const fileUploadModelOne: FileUploadModel = {
          displayName: "Document sample",
          id: responseOne.data.id,
          contentType: "application/pdf",
          name: fileName,
        };

        const fileUploadModelTwo: FileUploadModel = {
          displayName: "Document sample",
          id: responseTwo.data.id,
          contentType: "application/pdf",
          name: fileName,
        };

        // 3. For each participant on the flow, create one instance of ParticipantUserModel
        const participant: UsersParticipantUserModel = {
          name: "Jack Bauer",
          email: "jack.bauer@mailnator.com",
          identifier: "75502846369",
        };

        // 4. Create a FlowActionCreateModel instance for each action (signature or approval) in the flow
        const flowAction: FlowActionsFlowActionCreateModel = {
          type: FlowActionType.Signer,
          user: participant,
        };

        // 5. Send the document create request
        documentsApi
          .apiDocumentsPost({
            files: [fileUploadModelOne, fileUploadModelTwo],
            // 6. Set up the attribute "isEnvelope" as "true" and you MUST give a name to this envelope "envelopeName"
            isEnvelope: true,
            envelopeName: "Name",
            flowActions: [flowAction],
          })
          .then((res) => {
            // 7. Print result
            console.log("Document ", res.data[0].documentId, "created");
          });
      });
  });
